utils/transformers/index.js

```diff
@@ -30,3 +31,240 @@ export const transformStyle = ({ nodeName, nodeValue, fillProp }) => {
+  rect: ['width', 'height'],
+  stop: ['offset', 'stopColor'],
+  svg: ['viewBox', 'width', 'height'],
+  text: ['fontFamily', 'fontSize', 'fontWeight', 'textAnchor'],
+  use: ['href'],
+};
+
+export const postProcessAttributes = (attributes, props, node) => attributes;
+
+export const elementsMap = {
+  ['circle']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.circle,
+  },
+  ['defs']: {
+    allowedAttributes: [],
+    postProcessAttributes: (attributes, props, node) => ({}),
+  },
+  ['ellipse']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.ellipse,
+  },
+  ['g']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.g,
+  },
+  ['line']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.line,
+  },
+  ['linearGradient']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.linearGradient,
+  },
+  ['path']: {
+    allowedAttributes: ALLOWED_ATTRIBUTES.path,
+    postProcessAttributes: (attributes, props, node) => {
+      const { fill } = props || {};
+
+      if (!fill) {
+        return attributes;
+      }
+
+      return Object.assign({}, attributes, { fill });
+    },
+  },
+  ['polygon']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.polygon,
+  },
+  ['polyline']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.polyline,
+  },
+  ['radialGradient']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.radialGradient,
+  },
+  ['rect']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.rect,
+  },
+  ['stop']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.stop,
+  },
+  ['svg']: {
+    allowedAttributes: ALLOWED_ATTRIBUTES.svg,
+    postProcessAttributes: (attributes, props, node) => {
+      const { height, width } = props || {};
+
+      if (!height && !width) {
+        return attributes;
+      }
+
+      const attrs = Object.assign({}, attributes);
+
+      if (height) {
+        attrs.height = height;
+      }
+
+      if (width) {
+        attrs.width = width;
+      }
+
+      return attrs;
+    },
+  },
+  ['text']: {
+    postProcessAttributes,
+    allowedAttributes: ALLOWED_ATTRIBUTES.text,
+  },
+  ['tspan']: {
+    allowedAttributes: ALLOWED_ATTRIBUTES.text,
+    postProcessAttributes: (attributes, props, node) => {
+      let { y: attrY } = attributes || {};
+
+      if (!attrY) {
+        return attributes;
+      }
+
+      const y = getFixedYPosition(node, attrY);
+
+      return Object.assign({}, attributes, { y });
+    },
+  },
+  ['use']: {
+    allowedAttributes: ALLOWED_ATTRIBUTES.use,
+    postProcessAttributes: (attributes, props, node) => {
+      const href = getHrefValue(node);
+
+      return Object.assign({}, attributes, { href });
+    },
+  },
+};
+
+export const renderSvgElementByNodeWithItsChildNodes = (
```

Comment:
That's a great idea, but before we do that, let me see if I understood it correctly.

What would be the purpose of these tests? To make sure that the function is rendering only the allowed attributes or to ensure that each element has the expected allowed attributes?

I'm asking that, because depending on the answer, the implementation will be different.

Reply:
> What would be the purpose of these tests? To make sure that the function is rendering only the allowed attributes or to ensure that each element has the expected allowed attributes?

☝️ 
yeah, this function with the mapping above seems to be the core of this library. So having tests for it would would give more confidence for future maintenance.

Reply:
I was reviewing your comment about testing this function and realize it already has its tests, besides that, the functions on the map you mentioned has tests to ensure each element is being rendered properly.

Do you think we still need more tests? If so, what do you suggest to test?